Gtest/sensors_client.cpp

```diff
@@ -0,0 +1,167 @@
+        delete m_client_sock;
+        m_client_sock = nullptr;
+    }
+}
+
+void SensorsClient::vhal_connected_callback(SockClient *sock) {
+    ALOGI("connected to server successfully: %s, %d", sock->get_ip(), sock->get_port());
+    (void)(sock);
+    m_connected = true;
+}
+
+void SensorsClient::vhal_disconnected_callback(SockClient *sock) {
+    ALOGI("disconnected to server");
+    (void)(sock);
+    m_connected = false;
+}
+
+void SensorsClient::vhal_message_callback(SockClient* client) {
+    sensor_config_msg_t sensor_ctrl_msg;
+    char* pointer   = (char*)(&sensor_ctrl_msg);
+    int len         = sizeof(sensor_config_msg_t);
+    int retry_count = 30;
+    int left_size   = len;
+    while (left_size > 0) {
+        int ret = client->recv_data(pointer, left_size);
+        if (ret <= 0) {
+            if (errno == EINTR || errno == EWOULDBLOCK || errno == EAGAIN) {
+                usleep(1000);
+                if ((retry_count--) < 0) {
+                    ALOGW("[timeout], failed to recv sensor config data from vhal:target: %d, recved len: %d, [%s], \n", len, len - left_size, strerror(errno));
+                    return;
+                }
+                continue;
+            } else {
+                ALOGW("failed to recv sensor config data from vhal: %s\n", strerror(errno));
+                return;
+            }
+        }
+        left_size -= ret;
+        pointer += ret;
+    }
+
+    ALOGI("receive config message from sensor vhal, sensor type: %d, enabled: %d, sample period: %d",
+        sensor_ctrl_msg.sensor_type, sensor_ctrl_msg.enabled, sensor_ctrl_msg.sample_period);
+    if(sensor_ctrl_msg.sensor_type == SENSOR_TYPE_ACCELEROMETER) {
+        if(sensor_ctrl_msg.enabled) {
+            m_acc_enabled = true;
+        } else {
+            m_acc_enabled = false;
+        }
+    }
+    process_config_info(sensor_ctrl_msg);
+}
+
+int SensorsClient::get_payload_len(int sensor_type) {
+    int payload_len = 0;
+    switch (sensor_type) {
+        case SENSOR_TYPE_ACCELEROMETER:
+        case SENSOR_TYPE_GYROSCOPE:
+        case SENSOR_TYPE_MAGNETIC_FIELD:
+            payload_len = 3 * sizeof(float);
+            break;
+        case SENSOR_TYPE_ACCELEROMETER_UNCALIBRATED:
+        case SENSOR_TYPE_GYROSCOPE_UNCALIBRATED:
+        case SENSOR_TYPE_MAGNETIC_FIELD_UNCALIBRATED:
+            payload_len = 6 * sizeof(float);
+            break;
+        case SENSOR_TYPE_LIGHT:
+        case SENSOR_TYPE_PROXIMITY:
+        case SENSOR_TYPE_AMBIENT_TEMPERATURE:
+            payload_len = 1 * sizeof(float);
+            break;
+        default:
+            payload_len = 0;
+            ALOGW("unsupported sensor type %d", sensor_type);
+            break;
+    }
+    return payload_len;
+}
+
+
+void SensorsClient::process_config_info(sensor_config_msg_t& cfg) {
+    for(int i = 0; i < FAKA_DATA_LEN; i++) {
+        aic_sensors_event_t *fake_data = nullptr;
+        size_t pay_load_len = get_payload_len(cfg.sensor_type);
```

Comment:
Here if pay_load_len is zero than its an unsupported sensor type. We can add a check to ignore